File: main (1).c
/***************************

Name: Jada Johnson
Date: February 21, 2023
Tutor: Mrs. Penrose-Whittaker
Subject: Programming 1
Continuous Assessment 3 Question 1

***************************/

#include <stdio.h>
#include <stdbool.h>

int main() {
	// declares variable as characters
	char direction, L, R, D, U;

	printf("-- PROGRAMME USED TO DIRECT THE HONEY BADGER --\n");

    // loops until a valid character is entered
	while(true) {
		// accepts user's input
		printf("\n-- HONEY BADGER'S DIRECTIONS -- "
		       "\nL is Left \nR is Right \nU is Up \nD is Down "
		       "\n\nPlease enter the honey badger's direction: ");
		// leading space gets rid of any newline character from previous input
		scanf(" %c", &direction);

		// depends on user's input; if l is entered, lion, if r, then rattlesnake, if u, then uakari, and finally, if d, dhole
		if (direction == 'L' || direction == 'l') {
			printf("This direction leads you to the Badger vs Lion");
		} else if (direction == 'R' || direction == 'r') {
			printf("This direction leads you to the Badger vs Rattlesnake");
		} else if (direction == 'U' || direction == 'u') {
			printf("This direction leads you to the Badger vs Uakari");
		} else if (direction == 'D' || direction == 'd') {
			printf("This direction leads you to the Badger vs Dhole");
		} else {
			printf("\nERROR: Invalid direction! Please enter either L, R, U, or D.\n");
		}
	}

	return 0;
}

File: main (1).py
'''

Name: Jada Johnson
Tutor: Mrs. Penrose-Whittaker
Date: March 14, 2023
Subject: Programming 1
Question: Accept the GPA for each person in a class of 10 students. 
Find and display the average GPA for the class.
Test Data: 3.4, 2.3, 2.7, 4.5, 3.9, 3.3, 4.6, 3.0, 3.6, 1.5 = 3.28

'''
CLASS_SIZE = 10
overallGPA = 0

print ("-- PROGAMME TO DETERMINE A CLASS' AVERAGE GPA --")

# loops through students in the class size
for student in range(CLASS_SIZE):
    print ("\nProccessing Student", student+1)
    while True:
        # takes user input, and  and stores it in gpa
        gpa = float(input('Please enter the student\'s GPA: '))
            
        if 0.0 <= gpa <= 5.0:
            overallGPA += gpa # Adds student's gpa to the overall gpa on each iteration
            break  # exit the while loop and move to the next student
        else:
            print("\nERROR: Invalid input! Please enter a value between 0.0 and 5.0.")
        
# stores the average for print 
average = overallGPA / CLASS_SIZE
    
print ("\nThe average GPA of the class is", "{:.2f}".format(average))